File: TeeNew/Legend_Symbol.pas
unit Legend_Symbol;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series, TeeSurfa, TeCanvas, TeeTools;

type
  TLegendSymbol = class(TBaseForm)
    Label1: TLabel;
    Edit1: TEdit;
    UpDown1: TUpDown;
    Label2: TLabel;
    ComboBox1: TComboBox;
    Series1: TContourSeries;
    CheckBox1: TCheckBox;
    ChartTool1: TRotateTool;
    procedure FormCreate(Sender: TObject);
    procedure Edit1Change(Sender: TObject);
    procedure ComboBox1Change(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TLegendSymbol.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(10);
  Series1.ValueFormat:='0.000';
end;

procedure TLegendSymbol.Edit1Change(Sender: TObject);
begin
  inherited;
  Chart1.Legend.Symbol.Width:=UpDown1.Position
end;

procedure TLegendSymbol.ComboBox1Change(Sender: TObject);
begin
  inherited;
  if ComboBox1.ItemIndex=0 then Chart1.Legend.Symbol.Position:=spLeft
                           else Chart1.Legend.Symbol.Position:=spRight;
end;

procedure TLegendSymbol.CheckBox1Click(Sender: TObject);
begin
  Chart1.Legend.Symbol.Continuous:=CheckBox1.Checked
end;

procedure TLegendSymbol.FormShow(Sender: TObject);
begin
  inherited;
  ComboBox1.ItemIndex:=0;
end;

initialization
  RegisterClass(TLegendSymbol);
end.

File: TeeNew/Builder2007/Legend_Symbol.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Legend_Symbol.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TLegendSymbolForm *LegendSymbolForm;
//---------------------------------------------------------------------------
__fastcall TLegendSymbolForm::TLegendSymbolForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TLegendSymbolForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(10);
  Series1->ValueFormat = "0.000";
}
//---------------------------------------------------------------------------

void __fastcall TLegendSymbolForm::FormShow(TObject *Sender)
{
  ComboBox1->ItemIndex = 0;
}
//---------------------------------------------------------------------------

void __fastcall TLegendSymbolForm::Edit1Change(TObject *Sender)
{
  Chart1->Legend->Symbol->Width = UpDown1->Position;
}
//---------------------------------------------------------------------------

void __fastcall TLegendSymbolForm::ComboBox1Change(TObject *Sender)
{
  if (ComboBox1->ItemIndex == 0) Chart1->Legend->Symbol->Position = spLeft;
  else Chart1->Legend->Symbol->Position = spRight;

}
//---------------------------------------------------------------------------

void __fastcall TLegendSymbolForm::CheckBox1Click(TObject *Sender)
{
  Chart1->Legend->Symbol->Continuous = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

